updates manual with codegenDecl on params docs (#22333)
nim-lang/Nim: doc/manual.md

The `codegenDecl` pragma can be used to directly influence Nim's code
generator. It receives a format string that determines how the variable,

> proc, proc param, or object type is declared in the generated code.

Reviewer: ```suggestion
routine, routine parameter, or object type is declared in the generated code.
```

  __interrupt void myinterrupt()
  ```

> For params it works like for variables.

Reviewer 2: ```suggestion
Parameters are annotated like variables.
```

> The following nim code:

Reviewer: ```suggestion
The following Nim code:
```

```nim

> proc add(a {.codegenDecl:"const $#& $#".} : int) 

Reviewer: ```suggestion
proc add(a {.codegenDecl: "const $#& $#".}: int) 
```

```

will generate this code:

```c++

> void add( const NI& a)

Reviewer: ```suggestion
void add(const NI& a)
```
